### ISSUE TITLE
Multiple top-level packages discovered in a flat-layout - Vsphere SDK

### ISSUE THREAD

**Issue by danama** [bb-Ricardo/netbox-sync#160]
Hello,

While in venv during the installation, there is an issue with the Vsphere Automation SDK install. This is also happening if building the docker image and on another system.

Outside of the venv, nothing to report.

Here comes the logs :

```
Collecting git+https://github.com/vmware/vsphere-automation-sdk-python.git
  Cloning https://github.com/vmware/vsphere-automation-sdk-python.git to /tmp/pip-req-build-m46egoho
  Running command git clone --filter=blob:none --quiet https://github.com/vmware/vsphere-automation-sdk-python.git /tmp/pip-req-build-m46egoho
  Resolved https://github.com/vmware/vsphere-automation-sdk-python.git to commit d8ee747115e2ad7505692723c60765824104e3b7
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [14 lines of output]
      error: Multiple top-level packages discovered in a flat-layout: ['lib', 'samples', 'sample_template'].
      
      To avoid accidental inclusion of unwanted files or directories,
      setuptools will not proceed with this build.
      
      If you are trying to create a single distribution with multiple packages
      on purpose, you should not rely on automatic discovery.
      Instead, consider the following options:
      
      1. set up custom discovery (`find` directive with `include` or `exclude`)
      2. use a `src-layout`
      3. explicitly set `py_modules` or `packages` with a list of names
      
      To find more information, look for "package discovery" on setuptools docs.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

**Comment by bb-Ricardo** [comment]
Oh, thank you for this hint. I will check it out.

were you able to try the installation with an older version?

**Comment by rizlas** [comment]
This is setuptools related. I encountered this yesterday. Sticking with setuptools < 60 works as expected.
https://github.com/pypa/setuptools/issues/3197

**Comment by bb-Ricardo** [comment]
Ahh great, good find. I guess we need to wait until VMware adopted to this behaviour.

There is also an open bug: https://github.com/vmware/vsphere-automation-sdk-python/issues/303

**Comment by danama** [comment]
Just tested building the Dockerfile with setuptools locked to 6.10.0 and it worked. Thanks !

**Comment by bb-Ricardo** [comment]
This got temporarily fixed with e5d4225cad4efb1e46bc5e604b01cbceff6808e3

**Comment by bb-Ricardo** [comment]
now resolved in "development"
